javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -842,37 +816,39 @@ public int length() {
     @Override
     public <U> Vector<U> map(Function<? super T, ? extends U> mapper) {
         Objects.requireNonNull(mapper, "mapper is null");
-        HashArrayMappedTrie<Integer, U> trie = HashArrayMappedTrie.empty();
-        for (int i = 0; i < length(); i++) {
-            trie = trie.put(i, mapper.apply(get(i)));
+
+        if (isEmpty()) {
+            return empty();
+        } else {
+            final Iterator<? extends U> results = iterator().map(mapper);
+            return results.isEmpty() ? empty()
+                                     : ofAll(results);
         }
-        return wrap(trie);
```

Comment:
I believe this code is enough
```java
public <U> Vector<U> map(Function<? super T, ? extends U> mapper) {
    return ofAll(iterator().map(mapper));
}
```

Method `ofAll` returns singleton if parameter is empty